src/main/java/com/cleanroommc/groovyscript/compat/mods/advancedrocketry/BaseRegistry.java

```diff
@@ -0,0 +1,195 @@
+        Class<?> clazz = getMachineClass();
+        RecipesMachine registry = RecipesMachine.getInstance();
+        List<IRecipe> recipes = registry.getRecipes(clazz);
+        if (recipes == null) {
+            recipes = new LinkedList<>();
+            // NOTE: this cast is completely invalid but Advanced Rocketry's code is pepega
+            // @see zmaster587.libVulpes.recipe.RecipesMachine#recipeList
+            // @see zmaster587.advancedRocketry.block.BlockSmallPlatePress#getRecipe
+            registry.recipeList.put((Class<? extends TileMultiblockMachine>) clazz, recipes);
+        }
+        return recipes;
+    }
+
+    @Override
+    public void onReload() {
+        List<IRecipe> recipes = getRecipeList();
+        recipes.addAll(restoreFromBackup());
+        recipes.removeAll(removeScripted());
+    }
+
+    public void addRecipe(IRecipe r) {
+        getRecipeList().add(r);
+        addScripted(r);
+    }
+
+    public boolean removeRecipe(IRecipe r) {
+        if (getRecipeList().remove(r)) {
+            addBackup(r);
+            return true;
+        }
+        return false;
+    }
+
+    public boolean removeByFluidInput(FluidStack fluidStack) {
+        return getRecipeList().removeIf(r -> {
+            List<FluidStack> inputFluid = r.getFluidIngredients();
+            if (inputFluid.stream().anyMatch(fluidIn -> fluidIn.isFluidEqual(fluidStack))) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    public boolean removeByInput(IIngredient inputItem) {
+        if (inputItem instanceof FluidStack) {
+            return removeByFluidInput((FluidStack) inputItem);
+        }
+        return getRecipeList().removeIf(r -> {
+            List<List<ItemStack>> input = r.getIngredients();
+            if (input.stream().anyMatch(itemStacks -> itemStacks.stream().anyMatch(inputItem))) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    public boolean removeByFluidOutput(FluidStack fluidStack) {
+        return getRecipeList().removeIf(r -> {
+            List<FluidStack> outputFluid = r.getFluidOutputs();
+            if (outputFluid.stream().anyMatch(fluidOut -> fluidOut.isFluidEqual(fluidStack))) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    public boolean removeByOutput(IIngredient outputItem) {
+        if (outputItem instanceof FluidStack) {
+            return removeByFluidInput((FluidStack) outputItem);
+        }
+        return getRecipeList().removeIf(r -> {
+            List<ItemStack> output = r.getOutput();
+            if (output.stream().anyMatch(outputItem)) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(example = @Example(priority = 2000, commented = true))

```

Comment:
you are currently doing this: `@MethodDescription(example = @Example(priority = 2000, commented = true))`
this means that, inside this method description, there is a single example. this example is empty, commented, and sorted compared to other examples for this method with a priority of 2000.

almost all removeAll method descriptions do: `@MethodDescription(priority = 2000, example = @Example(commented = true))`
this means that this method description is sorted compared to other method descriptions with a priority of 2000, and has a single commented example.

the latter should be used, since it means the order of methods would be `removeByInput`, `removeByOutput`, `removeAll`, putting the "bulk removal" option at the end.

Reply:
I keep forgetting about this. Oopsie.